src/main/java/io/axoniq/axonserver/connector/command/impl/CommandChannelImpl.java

```diff
@@ -156,7 +157,13 @@ private void handleAck(CommandProviderInbound message, ReplyChannel<CommandProvi
     }
 
     @Override
-    public synchronized void connect() {
+    public void connect() {
+        if (!commandHandlers.isEmpty()) {
```

Comment:
I understand the addition of the `if`, but I feel the knowledge why this is done will eventually get lost in a year or so. Isn't there a clearer way to define this operation? Maybe move the check to a private method or so with a name specifying what's being validated?